materialize/resources/resource_connection.go

```diff
@@ -0,0 +1,838 @@
+				ForceNew:      true,
+				ConflictsWith: []string{"ssh_host", "ssh_user", "ssh_port"},
+				RequiredWith:  []string{"aws_privatelink_availability_zones"},
+			},
+			"aws_privatelink_availability_zones": {
+				Description:   "The availability zones of the AWS PrivateLink service.",
+				Type:          schema.TypeList,
+				Elem:          &schema.Schema{Type: schema.TypeString},
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"ssh_host", "ssh_user", "ssh_port"},
+				RequiredWith:  []string{"aws_privatelink_service_name"},
+			},
+			"postgres_database": {
+				Description: "The target Postgres database.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				RequiredWith: []string{
+					"postgres_host",
+					"postgres_port",
+					"postgres_user",
+					"postgres_password",
+				},
+			},
+			"postgres_host": {
+				Description: "The Postgres database hostname.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				RequiredWith: []string{
+					"postgres_database",
+					"postgres_port",
+					"postgres_user",
+					"postgres_password",
+				},
+			},
+			"postgres_port": {
+				Description: "The Postgres database port.",
+				Type:        schema.TypeInt,
+				Optional:    true,
+				RequiredWith: []string{
+					"postgres_database",
+					"postgres_host",
+					"postgres_user",
+					"postgres_password",
+				},
+			},
+			"postgres_user": {
+				Description: "The Postgres database username.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				RequiredWith: []string{
+					"postgres_database",
+					"postgres_host",
+					"postgres_port",
+					"postgres_password",
+				},
+			},
+			"postgres_password": {
+				Description: "The Postgres database password.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				RequiredWith: []string{
+					"postgres_database",
+					"postgres_host",
+					"postgres_port",
+					"postgres_user",
+				},
+			},
+			"postgres_ssh_tunnel": {
+				Description: "The SSH tunnel configuration for the Postgres database.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"postgres_ssl_ca": {
+				Description: "The CA certificate for the Postgres database.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"postgres_ssl_cert": {
+				Description: "The client certificate for the Postgres database.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"postgres_ssl_key": {
+				Description: "The client key for the Postgres database.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"postgres_ssl_mode": {
+				Description: "The SSL mode for the Postgres database.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"postgres_aws_privatelink": {
+				Description: "The AWS PrivateLink configuration for the Postgres database.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"kafka_broker": {
```

Comment:
Should we remove this and just use `kafka_brokers` and allow 1 or more?

Reply:
As we discussed the just discussed, this follows the [docs](https://materialize.com/docs/sql/create-connection/#general-options), we can later decide if we want to keep it or not.